template/workflows/helm/.github/workflows/azure-kubernetes-service-helm.yml

```diff
@@ -108,34 +109,19 @@ jobs:
           admin: 'false'
           use-kubelogin: 'true'
 
-      # Runs Helm to create manifest files
-      - name: Bake deployment
-        uses: azure/k8s-bake@v2
-        with:
-          renderEngine: "helm"
-          helmChart: ${{ env.CHART_PATH }}
-          overrideFiles: ${{ env.CHART_OVERRIDE_PATH }}
-          overrides: ${{ env.CHART_OVERRIDES }}
-          helm-version: "latest"
-        id: bake
-
       # Checks if the AKS cluster is private
       - name: Is private cluster
         id: isPrivate
         run: |
           result=$(az aks show --resource-group ${{ env.CLUSTER_RESOURCE_GROUP }} --name ${{ env.CLUSTER_NAME }} --query "apiServerAccessProfile.enablePrivateCluster")
           echo "PRIVATE_CLUSTER=$result" >> "$GITHUB_OUTPUT"
 
-      # Deploys application based on manifest files from previous step
-      - name: Deploy application
-        uses: Azure/k8s-deploy@v4
-        with:
-          action: deploy
-          manifests: ${{ steps.bake.outputs.manifestsBundle }}
-          images: |
-            ${{ env.AZURE_CONTAINER_REGISTRY }}.azurecr.io/${{ env.CONTAINER_NAME }}:${{ github.sha }}
-          resource-group: ${{ env.CLUSTER_RESOURCE_GROUP }}
-          name: ${{ env.CLUSTER_NAME }}
-          private-cluster: ${{ steps.isPrivate.outputs.PRIVATE_CLUSTER }}
-          namespace: ${{ env.NAMESPACE }}
+      # Deploys application
+      - name: Deploy application on private cluster
+        if : steps.isPrivate.outputs.PRIVATE_CLUSTER == 'true'
+        run: az aks command invoke --resource-group ${{ env.CLUSTER_RESOURCE_GROUP }} --name ${{ env.CLUSTER_NAME }} --command "helm upgrade --wait -i -f ${{ env.CHART_OVERRIDE_PATH }} --set ${{ env.CHART_OVERRIDES }} automated-deployment ${{ env.CHART_PATH }} --namespace ${{env.NAMESPACE}}" --file .
```

Comment:
should we compare the helm command output to see the actual exit code?

Reply:
The helm command keeps running even after the az aks command times out. So I couldn't find a way to capture the exit code of the helm command. I tried with the '[az aks command result](https://learn.microsoft.com/en-us/cli/azure/aks/command?view=azure-cli-latest#az-aks-command-result)' which is supposed to return the result of the previously triggered 'aks command invoke', but it didn't return the exit code of the helm command, since it was still running.

Reply:
can you set a timeout on the helm command?

Reply:
the default timeout is 5 minutes. we should be able to use az aks command result to figure it out based on that timeout with some bash.

Reply:
Good idea, i've set a timeout for 4 min for the helm command and then returning the corresponding exit code